powerdns_recursor/datadog_checks/powerdns_recursor/powerdns_recursor.py

```diff
@@ -126,6 +126,27 @@ def check(self, instance):
                     self.gauge('powerdns.recursor.{}'.format(stat['name']), float(stat['value']), tags=tags)
                 elif stat['name'] in PowerDNSRecursorCheck.RATE_METRICS_V4:
                     self.rate('powerdns.recursor.{}'.format(stat['name']), float(stat['value']), tags=tags)
+        self._collect_metadata(config)
+
+    def _collect_metadata(self, config):
```

Comment:
Consider marking this as the metadata entrypoint (skips calling the method if `enable_metadata_collection` is disabled in the Agent):

```suggestion
    @AgentCheck.metadata_entrypoint
    def _collect_metadata(self, config):
```

Refs #6084

Will add this to https://datadoghq.dev/integrations-core/base/api/#datadog_checks.base.checks.base.AgentCheck